fendermint/vm/interpreter/src/fvm/upgrades.rs

```diff
@@ -63,17 +61,26 @@ where
         new_app_version: Option<u64>,
         migration: MigrationFunc<DB>,
     ) -> anyhow::Result<Self> {
-        let mut upgrade = Self {
-            chain_name: chain_name.to_string(),
-            chain_id: 0.into(),
+        Ok(Self {
+            chain_id: chainid::from_str_hashed(&chain_name.to_string())?,
```

Comment:
Any reason we don't bind the upgrade to the chain name instead?

Reply:
@raulk Internally in `FvmMessageInterpreter` we [compare using chain_id](https://github.com/consensus-shipyard/ipc/blob/main/fendermint/vm/interpreter/src/fvm/exec.rs#L216). We could compare it using the chain name if I can somehow access its value there :S

Reply:
We don't remember the chain name after genesis.